src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    const isHerbivore = animal instanceof Herbivore;
+
+    if (!isHerbivore || animal.hidden) {
+      return;
+    }
+
+    animal.health -= 50;
+
+    Animal.alive = Animal.alive.filter(beast => beast.health !== 0);
```

Comment:
Lets move this to Animal class and put inside new method e.g. die and then reuse it here 
```suggestion
    animal.die();
```